test_conformance/extensions/cl_khr_command_buffer/command_buffer_printf.cpp

```diff
@@ -0,0 +1,556 @@
+
+        error = clSetKernelArg(kernel, 2, sizeof(off_mem), &off_mem);
+        test_error(error, "clSetKernelArg failed");
+
+        return CL_SUCCESS;
+    }
+
+    //--------------------------------------------------------------------------
+    cl_int SetUp(int elements) override
+    {
+        // Query if device supports simultaneous use
+        cl_device_command_buffer_capabilities_khr capabilities;
+        cl_int error =
+            clGetDeviceInfo(device, CL_DEVICE_COMMAND_BUFFER_CAPABILITIES_KHR,
+                            sizeof(capabilities), &capabilities, NULL);
+        test_error(
+            error,
+            "Unable to query CL_COMMAND_BUFFER_CAPABILITY_KERNEL_PRINTF_KHR");
+
+        if ((capabilities & CL_COMMAND_BUFFER_CAPABILITY_SIMULTANEOUS_USE_KHR)
+            == 0)
+        {
+            log_error(
+                "Device capability "
+                "CL_COMMAND_BUFFER_CAPABILITY_KERNEL_PRINTF_KHR not supported");
+            return CL_INVALID_DEVICE_TYPE;
+        }
+
+        temp_filename = get_temp_filename();
+        if (temp_filename.empty())
+        {
+            log_error("get_temp_filename failed\n");
+            return -1;
+        }
+
+        return BasicCommandBufferTest::SetUp(elements);
+    }
+
+    //--------------------------------------------------------------------------
+    cl_int Run() override
+    {
+        cl_int error = CL_SUCCESS;
+
+        // record command buffer with primary queue
+        error = RecordCommandBuffer();
+        test_error(error, "RecordCommandBuffer failed");
+
+        if (simultaneous_use_support)
+        {
+            // enque simultaneous command-buffers with printf calls
+            error = RunSimultaneous();
+            test_error(error, "RunSimultaneous failed");
+        }
+        else
+        {
+            // enque single command-buffer with printf calls
+            error = RunSingle();
+            test_error(error, "RunSingle failed");
+        }
+
+        std::remove(temp_filename.c_str());
+
+        return CL_SUCCESS;
+    }
+
+    //--------------------------------------------------------------------------
+    cl_int RecordCommandBuffer()
+    {
+        cl_int error = CL_SUCCESS;
+
+        error = clCommandNDRangeKernelKHR(
+            command_buffer, nullptr, nullptr, kernel, 1, nullptr, &num_elements,
+            nullptr, 0, nullptr, nullptr, nullptr);
+        test_error(error, "clCommandNDRangeKernelKHR failed");
+
+        error = clFinalizeCommandBufferKHR(command_buffer);
+        test_error(error, "clFinalizeCommandBufferKHR failed");
+        return CL_SUCCESS;
+    }
+
+    //--------------------------------------------------------------------------
+    int WaitForEvent(cl_event* event)
+    {
+        cl_int status = clWaitForEvents(1, event);
+        if (status != CL_SUCCESS)
+        {
+            log_error("clWaitForEvents failed");
+            return status;
+        }
```

Comment:
Think this could be `test_error(status, "clWaitForEvents failed")`

Reply:
Method removed due to switch from cl_event to clEventWrapper.